### PR TITLE
[Website] Highlights section backend integration

### PR DESCRIPTION
**Pull request by FaithDaka**
#### Summary of Changes (What does this PR do?)
- A clean up of the highlights integration chaotic PR. _Refer to it for screenshots._

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):
- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* cd website
* git checkout staging
* git pull origin staging
* Update .env file with the values of variables: _SECURE_SSL_REDIRECT_ and _HTTP_X_FORWARDED_PROTO_
* npm install (node v12)
* Open two terminals: one to run server and other to run client
* In client terminal run: 
- * npm run build 
- * npm run serve

* In server terminal, activate a virtual environment to load env variables and run: 

- * python manage.py makemigrations
- * python manage.py migrate
- * python manage.py collectstatic
- * python manage.py runserver

* Website runs on port: 8000. Route to the admin portal to test the Highlights DB and view changes in the frontend.

#### What are the relevant tickets?
- [WEB-295]

[WEB-295]: https://airqoteam.atlassian.net/browse/WEB-295?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/backend/highlights/serializers.py [hunk 1]
```diff
@@ -5,6 +5,11 @@ class HighlightSerializer(serializers.ModelSerializer):
     class Meta:
         model = Highlight
         fields = '__all__'
+    image = serializers.SerializerMethodField()
+
+    @staticmethod
+    def get_image(obj):
+        return obj.image.url
```
Comment (FaithDaka) [inline website/backend/highlights/serializers.py:12]: Get image from cloudinary and show it in frontend

File: website/backend/highlights/views.py [hunk 2]
```diff
@@ -1,11 +1,14 @@
 from rest_framework import viewsets
+from rest_framework.permissions import AllowAny
 from .models import Highlight, Tag
 from .serializers import HighlightSerializer, TagSerializer
 
 class HighlightViewSet(viewsets.ReadOnlyModelViewSet):
+    permission_classes = (AllowAny,)
```
Comment (FaithDaka) [inline website/backend/highlights/views.py:7]: Allow access to GET Api without need for an access token

File: website/frontend/apis/index.js [hunk 3]
```diff
@@ -30,3 +31,7 @@ export const getAllDepartmentsApi = async () => await axios.get(DEPARTMENTS_URL)
 
 // Teams endpoints
 export const getAllTeamMembersApi = async () => await axios.get(TEAMS_URL).then((response) => response.data);
+
+// Highlights endpoints
+export const getAllHighlightsApi = async () => await axios.get(HIGHLIGHTS_URL).then((response)=>(response.data));
+export const getAllTagsApi = async () => await axios.get(TAGS_URL).then((response) => (response.data));
```
Comment (FaithDaka) [inline website/frontend/apis/index.js:37]: Get Highlights and Tags from API

File: website/frontend/config/urls.js [hunk 4]
```diff
@@ -32,6 +32,13 @@ export const CAREERS_URL = `${WEBSITE_BASE_URL}/career/`;
...
+// members urls
 export const TEAMS_URL = `${WEBSITE_BASE_URL}/team/`;
 
+// netmanager url
 export const NETMANAGER_URL = stripTrailingSlash(process.env.REACT_NETMANAGER_BASE_URL.replace("/api/v1/", ""));
+
+// highlights urls
+export const HIGHLIGHTS_URL = `${WEBSITE_BASE_URL}/highlights/`;
+
+export const TAGS_URL = `${WEBSITE_BASE_URL}/tags/`;
```
Comment (FaithDaka) [inline website/frontend/config/urls.js:44]: End-point URLs

File: website/frontend/reduxStore/Highlights/operations.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { isEmpty } from 'underscore';
```
Comment (FaithDaka) [inline website/frontend/reduxStore/Highlights/operations.js:1]: Add data from API to the Redux store

File: website/frontend/src/components/HighlightsSection/index.js [hunk 6]
```diff
@@ -1,72 +1,88 @@
-import React, { useState } from 'react'
```
Comment (FaithDaka) [inline website/frontend/src/components/HighlightsSection/index.js:1]: Update HighlightsSection frontend

File: website/frontend/src/components/LoadSpinner/ImageLoader.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
```
Comment (FaithDaka) [inline website/frontend/src/components/LoadSpinner/ImageLoader.js:1]: Added load-spinner for an image loading

File: website/frontend/src/components/LoadSpinner/SectionLoader.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
Comment (FaithDaka) [inline website/frontend/src/components/LoadSpinner/SectionLoader.js:1]: Added load-spinner for a section loading

File: website/frontend/src/pages/CareerPage.js [hunk 9]
```diff
@@ -1,115 +1,153 @@
 import React, { useState, useEffect } from 'react';
```
Comment (FaithDaka) [inline website/frontend/src/pages/CareerPage.js:1]: Updates to careers page

**Comment by Baalmart** [comment]
Hi @FaithDaka , on the _client terminal_, I am experiencing some small challenge as shared below on running `npm build`. Care to shed some light?
`npm WARN build `npm build` called with no arguments. Did you mean to `npm run-script build`?`

**Comment by FaithDaka** [comment]
Ohh yes. npm requires the key-word: **run** to run scripts
The command is **npm run build**. Sorry about that @Baalmart

**Comment by Baalmart** [comment]
@FaithDaka 
On running migrate, I get quite a verbose of error messages but managed to pick put just one significant one shared below.

**django.core.exceptions.ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.**

**Comment by FaithDaka** [comment]
You'll need to setup a postgres database and provide that to the DATABASE_URI variable in the .env
The Readme has insight on that.
For updated insight on the process: https://github.com/airqo-platform/AirQo-frontend/pull/750

**Comment by Baalmart** [comment]
@Faith Daka
Three last feedback items:

- PR description seems to miss the `pip install` instruction, we could edit this description for other reviewers.  
- On running python manage.py collectstatic, I get` Unknown command: 'collectstatic'`
- On running `python manage.py runserver,` I get this error: **CommandError: You must set settings.ALLOWED_HOSTS if DEBUG is False.**

Thanks again for this AMAZING work!

**Comment by FaithDaka** [comment]
The assumption is the READMe is read first before any interactions (PR Reviews) with the website. To make sure all requirements for the website are up to date, to avoid specifying at PR level @Baalmart. Thanks

**Comment by Baalmart** [comment]
> The assumption is the READMe is read first before any interactions (PR Reviews) with the website. This ensure all requirements for the website are up to date, and avoids the re-specifying at PR level @Baalmart. Thanks

Alright @FaithDaka , thanks for the clarification on this. I will give it a second try as soon as possible.

File: website/frontend/src/pages/CareerPage.js (hunk 9 above)
Comment (Baalmart) [reply website/frontend/src/pages/CareerPage.js:1]: Thanks @FaithDaka for these PR review comments, much appreciated and welcome! Keep it up!
Just a few comments:

1. Could we add some more detail? What are we updating on the career's page? Are we consuming the backend endpoints/service on the careers page? etc? etc?
2. Secondly, there seems to be some new conflicts that could have been brought about by the recently merged [Website SEO PR ](https://github.com/airqo-platform/AirQo-frontend/pull/761) by @Codebmk 

Thanks!

**Review by Baalmart — APPROVED** [review]
LGTM, thanks @FaithDaka

**Pull request merged**